File: agent-driven-architecture-main/examples/typescript/src/blocks/console/tools.ts
// ── blocks/console/tools — presentation verbs, ONE tool mechanic (6.8) ─────
// Compare line for line with `blocks/triage/tools.ts`. Same declarations, same
// `sign` entry, same reversibility decision. The authoring discipline is
// identical (4.4) because there is only one discipline.

import type { Verb } from "@adr/spine/pure/verb";
import { reversible } from "@adr/spine/pure/verb";
import { boolean, object, string } from "valibot";
import type {
  FocusTicketCommand,
  FocusTicketResult,
  SetPanelCommand,
  SetPanelResult,
} from "./contract";

export function consoleVerbs<S>(): readonly Verb<S>[] {
  return [
    reversible<S, { ticket: string }, FocusTicketResult, FocusTicketCommand>({
      name: "focusTicket",
      describe: "Bring a ticket into focus on the console.",
      schema: object({ ticket: string() }),
      run: (input) => ({ outcome: "ok", tool: "focusTicket", ticket: input.ticket }),
      sign: (result, sig, id) => ({
        outcome: "ok",
        tool: "focusTicket",
        sig,
        id,
        ticket: result.ticket,
      }),
    }),
    reversible<S, { panel: string; visible: boolean }, SetPanelResult, SetPanelCommand>({
      name: "setPanel",
      describe: "Show or hide a console panel (e.g. 'escalation', 'findings').",
      schema: object({ panel: string(), visible: boolean() }),
      run: (input) => ({
        outcome: "ok",
        tool: "setPanel",
        panel: input.panel,
        visible: input.visible,
      }),
      sign: (result, sig, id) => ({
        outcome: "ok",
        tool: "setPanel",
        sig,
        id,
        panel: result.panel,
        visible: result.visible,
      }),
    }),
  ];
}

File: agent-driven-architecture-main/examples/kotlin/block/console/src/main/kotlin/adr/blocks/console/Tools.kt
// ── blocks/console/tools — presentation verbs, authored EXACTLY like domain ones ─
// Same five properties, same signer, same registry, same four sites (§11.1). Put
// this file beside blocks/triage/tools and read them together: there is no
// difference to find, which is the point of 6.8.

package adr.blocks.console

import adr.contract.ConsoleCommand
import adr.contract.ConsoleResult
import adr.spine.pure.Lens
import adr.spine.pure.PanelId
import adr.spine.pure.RawInput
import adr.spine.pure.TicketId
import adr.spine.pure.ToolName
import adr.spine.pure.Verb

val FOCUS_TICKET = ToolName("focusTicket")
val SET_PANEL = ToolName("setPanel")

internal data class FocusInput(val ticket: TicketId)

internal data class PanelInput(val panel: PanelId, val visible: Boolean)

internal class ConsoleTools<S>(private val lens: Lens<S, ConsoleSlice>) {

    fun verbs(): List<Verb<S, *, *>> = listOf(
        Verb.Reversible(
            name = FOCUS_TICKET,
            describe = "Bring a ticket into focus on the console.",
            decode = ::decodeFocus,
            run = { input, _ -> ConsoleResult.FocusTicket(FOCUS_TICKET, input.ticket) },
            sign = { r, sig, id -> ConsoleCommand.FocusTicket(r.tool, sig, id, r.ticket) },
            narrow = { it as? ConsoleResult.FocusTicket },
        ),
        Verb.Reversible(
            name = SET_PANEL,
            describe = "Show or hide a console panel (queue | detail | audit).",
            decode = ::decodePanel,
            run = { input, _ -> ConsoleResult.SetPanel(SET_PANEL, input.panel, input.visible) },
            sign = { r, sig, id -> ConsoleCommand.SetPanel(r.tool, sig, id, r.panel, r.visible) },
            narrow = { it as? ConsoleResult.SetPanel },
        ),
    )

    private fun decodeFocus(raw: RawInput): FocusInput? =
        raw.text("ticket")?.let { FocusInput(TicketId(it)) }

    private fun decodePanel(raw: RawInput): PanelInput? {
        val panel = raw.text("panel") ?: return null
        val visible = raw.flag("visible") ?: return null
        return PanelInput(PanelId(panel), visible)
    }
}
